File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int n = nums.size();
        int x = n/2;
        int temp ;
        int cnt =0;
        for(int i = 0 ; i < n ; i++){
            if(cnt == 0){
                cnt = 1;
                temp = nums[i];
            }else if(temp == nums[i]){
                cnt++;
            }else {
                cnt--;
            }
        }
        int cnt1 = 0;
        for(int i = 0; i<n; i++){
            if(nums[i]==temp)cnt1++;
        }
       if (cnt1 > (n / 2)) return temp;
    return -1;
    }
};

File: 0169-majority-element/0169-majority-element.rs
impl Solution {
    pub fn majority_element(nums: Vec<i32>) -> i32 {
        let mut n = nums.len();
        let mut el = nums[0];
      let mut count = 0;
     for &i in &nums {
         
         if count == 0{
             el = i;
         }
         if i == el {
             count +=1;
         }
         else{
             count -=1;
         }
          
        
         
        }
     el
    }
}